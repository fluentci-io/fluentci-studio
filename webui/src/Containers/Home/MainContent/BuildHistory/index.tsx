import BuildHistory from "./BuildHistory";

export default BuildHistory;
