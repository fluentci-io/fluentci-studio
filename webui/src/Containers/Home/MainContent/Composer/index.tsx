import Composer from "./ComposerWithData";

export default Composer;
