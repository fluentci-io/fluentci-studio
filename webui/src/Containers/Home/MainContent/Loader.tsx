import { FC } from "react";
import ContentLoader from "react-content-loader";

const Loader: FC = (props) => {
  return (
    <ContentLoader
      viewBox="0 0 778 116"
      backgroundColor="#1207326a"
      foregroundColor="#21163f"
      {...props}
    >
      <rect x="37" y="34" rx="0" ry="0" width="0" height="0" />
      <rect x="28" y="29" rx="0" ry="0" width="258" height="32" />
      <rect x="28" y="71" rx="0" ry="0" width="465" height="32" />
      <rect x="434" y="94" rx="0" ry="0" width="0" height="0" />
      <rect x="29" y="116" rx="0" ry="0" width="749" height="32" />
    </ContentLoader>
  );
};

export default Loader;
