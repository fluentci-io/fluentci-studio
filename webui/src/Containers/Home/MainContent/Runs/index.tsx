import Runs from "./RunsWithData";

export default Runs;
