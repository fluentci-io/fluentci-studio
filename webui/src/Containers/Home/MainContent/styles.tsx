import styled from "@emotion/styled";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 90vw;
  min-width: 1055px;
  margin: 0 auto;
  margin-top: 60px;

  @media (min-width: 1400px) {
    width: 85vw;
  }
`;

export const Title = styled.div`
  font-size: 20px;
  margin-bottom: 20px;
  font-weight: 500;
  flex: 1;
  font-family: "Lexend";
  color: #fff;
`;

export const Header = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 50px;
`;

export const RunButton = styled.button`
  font-family: "Lexend";
  height: 40px;
  background-color: #24ffb5;
  color: #000;
  border: none;
  font-weight: 600;
  width: 150px;
  cursor: pointer;
  &:hover {
    background-color: #18d193;
  }
`;

export const PopoverButton = styled.button`
  width: 40px;
  height: 40px;
  cursor: pointer;
  background-color: initial;
  border: none;
  color: #fff;
  margin-right: 10px;
`;

export const PictureWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 50px;
  width: 50px;
  // background-color: #33909933;
  border-radius: 8px;
  margin-right: 15px;
`;

export const Picture = styled.img`
  height: 34px;
  width: 34px;
`;

export const ProjectWrapper = styled.div`
  color: #fff;
  background-color: #10072c;
  height: 80px;
  border: 1px solid #ffffff1f;
  border-radius: 6px;
  padding: 20px;
  display: flex;
  flex-direction: row;
  align-items: center;
  &:hover {
    border: 1px solid #06ffe27d;
  }
`;

export const Path = styled.div`
  font-size: 12.5px;
  color: #06ffe0;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  width: 100%;
  margin-bottom: 10px;
`;

export const Row = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const Visibility = styled.div`
  color: #7392b1;
  font-size: 12px;
  margin-left: 10px;
  margin-top: 2px;
  border: 1px solid #7392b161;
  padding: 3px;
  padding-left: 6px;
  padding-right: 6px;
  border-radius: 14px;
`;

export const Archive = styled(Visibility)`
  color: #ff6600;
  border: 1px solid #ff6600;
  font-weight: 500;
  height: 18px;
`;

export const Tag = styled.button`
  font-size: 11px;
  color: #00f9ed;
  background-color: #00f9ec1c;
  padding: 4px;
  padding-left: 6px;
  padding-right: 6px;
  border-radius: 14px;
  margin-right: 5px;
  border: none;
  cursor: pointer;
  font-family: "Lexend";
`;

export const Avatar = styled.img`
  height: 60px;
  width: 60px;
  border-radius: 30px;
`;

export const Separator = styled.div`
  margin-top: 64px;
  margin-bottom: 48px;
  border-bottom: 1px solid #1c134fc3;
`;

export const GithubUserName = styled.div`
  font-size: 16px;
  color: rgba(115, 146, 177, 0.7);
  margin-left: 5px;
`;

export const LinkGithubRepo = styled.a`
  font-family: monospace;
  color: #fff;
  &:hover {
    text-decoration: underline;
  }
`;

export default {
  Tab: {
    Tab: {
      style: ({
        $isActive,
        $theme,
      }: {
        $isActive: string;
        $theme: { primaryFontFamily: string };
      }) => ({
        backgroundColor: "#0f0124 !important",
        color: $isActive ? "#24ffb5" : "#fff",
        fontFamily: $theme.primaryFontFamily,
        fontSize: "16px",
        ":hover": {
          color: "#24ffb5",
          backgroundColor: "#0f0124",
        },
      }),
    },
  },
  Tabs: {
    TabHighlight: {
      style: {
        backgroundColor: "#24ffb5",
      },
    },
    TabBorder: {
      style: {
        height: "0px",
      },
    },
  },
  StatefulMenu: {
    Option: {
      props: {
        overrides: {
          ListItem: {
            style: ({ $theme }: { $theme: { primaryFontFamily: string } }) => ({
              fontFamily: $theme.primaryFontFamily,
            }),
          },
        },
      },
    },
    List: {
      style: {
        border: "1px solid rgb(71 5 94 / 29%)",
      },
    },
  },
};
