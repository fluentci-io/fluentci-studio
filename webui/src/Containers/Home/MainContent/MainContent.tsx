import { FC } from "react";
import {
  Container,
  RunButton,
  Header,
  Title,
  ProjectWrapper,
  Picture,
  PictureWrapper,
  Path,
  Row,
  Visibility,
  Tag,
} from "./styles";
import { Project } from "../../../Hooks/GraphQL";
import { Link } from "react-router-dom";
import BuildHistory from "./BuildHistory";
import _ from "lodash";

export type MainContentProps = {
  projects?: Project[];
  onNewProject: () => void;
  displayNewProjectButton: boolean;
};

const MainContent: FC<MainContentProps> = (props) => {
  const { projects, onNewProject } = props;
  return (
    <div>
      <Container>
        <Header>
          <Title>Projects</Title>
          {props.displayNewProjectButton && (
            <RunButton onClick={onNewProject}>New Project</RunButton>
          )}
        </Header>
        {projects!.map((item, index) => (
          <div key={index} style={{ marginBottom: 25, position: "relative" }}>
            <ProjectWrapper>
              <PictureWrapper>
                <Picture src={item.picture} />
              </PictureWrapper>
              <div style={{ width: "calc(50% - 40px)" }}>
                <Link to={`/project/${item.id}`} style={{ color: "#fff" }}>
                  <Row>
                    <div>{item.displayName || item.name}</div>
                    {item.isPrivate === false && (
                      <Visibility>Public</Visibility>
                    )}
                  </Row>
                  {item.path !== "empty" && !!item.path && (
                    <Path>{item.path}</Path>
                  )}
                </Link>
                <div>
                  {item.tags?.map((tag) => (
                    <Tag>{tag}</Tag>
                  ))}
                </div>
              </div>
              {_.get(item, "recentRuns.0.status") && (
                <BuildHistory
                  status={_.last(item.recentRuns)?.status || "PENDING"}
                  reliability={item.reliability || 0}
                  speed={item.speed || 0}
                  buildsPerWeek={item.buildsPerWeek || 0}
                  builds={
                    Array.from(Array(18).keys()).map((i) => ({
                      status: _.get(item, `recentRuns.${i}.status`, ""),
                      duration: _.get(item, `recentRuns.${i}.duration`, 0),
                    })) || []
                  }
                />
              )}
            </ProjectWrapper>
          </div>
        ))}
      </Container>
      <div style={{ height: 100 }} />
    </div>
  );
};

MainContent.defaultProps = {
  projects: [],
};

export default MainContent;
