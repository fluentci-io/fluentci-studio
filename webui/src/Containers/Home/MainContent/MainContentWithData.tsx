import { FC } from "react";
import MainContent from "./MainContent";
import { useRecoilValue } from "recoil";
import { sideNavigationState } from "../../../Components/SideNavigation/SideNavigationState";
import {
  useGetJobQuery,
  useGetProjectQuery,
  useRunJobMutation,
  useRunPipelineMutation,
} from "../../../Hooks/GraphQL";

const MainContentWithData: FC = () => {
  const currentTab = useRecoilValue(sideNavigationState);
  const { data: jobData } = useGetJobQuery({
    variables: {
      id: "1",
    },
  });
  const { data: projectData } = useGetProjectQuery({
    variables: {
      id: "1",
    },
  });
  const [runJobMutation] = useRunJobMutation();
  const [runPipelineMutation] = useRunPipelineMutation();

  console.log(">> job", jobData);
  console.log(">> project", projectData);
  console.log(">> currentTab", currentTab);

  const onRun = async (id: string) => {
    console.log(">> onRun", id);
    await runJobMutation({
      variables: {
        jobName: "",
      },
    });
    await runPipelineMutation({
      variables: {
        projectId: "",
      },
    });
  };

  return (
    <MainContent
      title={projectData?.project?.name || ""}
      onRun={onRun}
      id={"1"}
    />
  );
};

export default MainContentWithData;
