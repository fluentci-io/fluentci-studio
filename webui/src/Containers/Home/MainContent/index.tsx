import MainContent from "./MainContentWithData";

export default MainContent;
