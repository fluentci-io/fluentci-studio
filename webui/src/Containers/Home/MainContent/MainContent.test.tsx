import { render } from "@testing-library/react";
import MainContent from "./MainContent";
import { projects } from "./mocks";
import { BrowserRouter } from "react-router-dom";

describe("MainContent", () => {
  it("should render the component", () => {
    const { container } = render(
      <MainContent
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        projects={projects as any[]}
        onNewProject={vi.fn()}
        displayNewProjectButton={true}
        loading={false}
      />,
      {
        wrapper: BrowserRouter,
      }
    );
    expect(container).toMatchSnapshot();
  });
});
