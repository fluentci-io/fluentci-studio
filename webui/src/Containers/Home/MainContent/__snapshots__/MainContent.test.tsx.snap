// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainContent > should render the component 1`] = `
<div>
  <div>
    <div
      class="css-14g8hm4"
    >
      <div
        class="css-1kk1ti"
      >
        <div
          class="css-1lqkj92"
        >
          Projects
        </div>
      </div>
      <div
        style="margin-bottom: 25px; position: relative;"
      >
        <div
          class="css-tg1f9m"
        >
          <div
            class="css-hg9guf"
          >
            <img
              class="css-18ewgrp"
              src="https://img.icons8.com/color-glass/96/maple-leaf.png"
            />
          </div>
          <div
            style="width: calc(50% - 40px);"
          >
            <a
              href="/project/hlv89kwqiwkkcizuhhz2vjwf"
              style="color: rgb(255, 255, 255);"
            >
              <div
                class="css-14zx4vh"
              >
                <div>
                  stoic-leavitt
                </div>
              </div>
              <div
                class="css-dz1vxl"
              >
                /home/tsirysndr/Documents/github/base-pipeline/example
              </div>
            </a>
            <div />
          </div>
          <div
            class="css-mxqn1d"
          >
            <div
              class="css-12j1nmi"
            >
              <div
                style="position: relative; width: 179px;"
              >
                <div
                  class="css-15vwjnl"
                  max="13.698697833016233"
                />
                <div
                  class="css-19zlhna"
                  max="6.232446278405114"
                />
                <div
                  class="css-1hm00j7"
                  max="7.233986475912793"
                />
                <div
                  class="css-1je50oj"
                  max="100"
                />
                <div
                  class="css-1rckjuv"
                  max="38.15901097905496"
                />
                <div
                  class="css-5uuauc"
                  max="15.191948143938458"
                />
                <div
                  class="css-1g90bfm"
                  max="5.592481859438446"
                />
                <div
                  class="css-148582a"
                  max="0"
                />
                <div
                  class="css-yyu4jy"
                  max="0"
                />
                <div
                  class="css-1fqssq5"
                  max="0"
                />
                <div
                  class="css-klpgw9"
                  max="0"
                />
                <div
                  class="css-1crqmc2"
                  max="0"
                />
                <div
                  class="css-1xyk97i"
                  max="0"
                />
                <div
                  class="css-1pdbres"
                  max="0"
                />
                <div
                  class="css-1ciawv4"
                  max="0"
                />
                <div
                  class="css-14s0ibw"
                  max="0"
                />
                <div
                  class="css-1vmcux9"
                  max="0"
                />
                <div
                  class="css-1hj9s5a"
                  max="0"
                />
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Speed
              </div>
              <div
                class="css-1o76smw"
              >
                32.28
                <span
                  class="css-n3629o"
                >
                  s
                </span>
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Reliability
              </div>
              <div
                class="css-1o76smw"
              >
                71.43
                <span
                  class="css-n3629o"
                >
                  %
                </span>
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Builds
              </div>
              <div
                class="css-1o76smw"
              >
                2
                <span
                  class="css-n3629o"
                >
                  / week
                </span>
              </div>
            </div>
            <div
              class="css-1gtb97y"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-sc-ea9ulj-0 jvZCGA"
                color="#05f076"
                fill="currentColor"
                focusable="false"
                height="24"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 25px; position: relative;"
      >
        <div
          class="css-tg1f9m"
        >
          <div
            class="css-hg9guf"
          >
            <img
              class="css-18ewgrp"
              src="https://img.icons8.com/color-glass/96/transportation.png"
            />
          </div>
          <div
            style="width: calc(50% - 40px);"
          >
            <a
              href="/project/w92rv9wl2lmir1106ds8sz8m"
              style="color: rgb(255, 255, 255);"
            >
              <div
                class="css-14zx4vh"
              >
                <div>
                  agitated-williams
                </div>
              </div>
              <div
                class="css-dz1vxl"
              >
                /home/tsirysndr/Documents/github/bun-pipeline/example
              </div>
            </a>
            <div />
          </div>
          <div
            class="css-mxqn1d"
          >
            <div
              class="css-12j1nmi"
            >
              <div
                style="position: relative; width: 179px;"
              >
                <div
                  class="css-1elkcgv"
                  max="28.285136679265182"
                />
                <div
                  class="css-dggfwz"
                  max="32.378189915326246"
                />
                <div
                  class="css-rr8ap1"
                  max="26.708857050774956"
                />
                <div
                  class="css-vo7k9s"
                  max="0"
                />
                <div
                  class="css-lb5wnk"
                  max="26.43637748674226"
                />
                <div
                  class="css-swc893"
                  max="36.13430605607805"
                />
                <div
                  class="css-15r98pb"
                  max="26.47446602795113"
                />
                <div
                  class="css-114xgcn"
                  max="30.626117019718144"
                />
                <div
                  class="css-1hjg8f2"
                  max="28.132782514429696"
                />
                <div
                  class="css-t7v4q8"
                  max="31.382028068324985"
                />
                <div
                  class="css-kt6z3u"
                  max="21.6694500600627"
                />
                <div
                  class="css-1ah38ok"
                  max="15.209047493481"
                />
                <div
                  class="css-9q6ihh"
                  max="15.663180100202162"
                />
                <div
                  class="css-18cjvli"
                  max="98.78995634467199"
                />
                <div
                  class="css-1mjocox"
                  max="58.33406580527966"
                />
                <div
                  class="css-1g1996u"
                  max="100"
                />
                <div
                  class="css-1r5m1px"
                  max="45.19351908821892"
                />
                <div
                  class="css-13mqueg"
                  max="59.453282939263424"
                />
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Speed
              </div>
              <div
                class="css-1o76smw"
              >
                12.91
                <span
                  class="css-n3629o"
                >
                  s
                </span>
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Reliability
              </div>
              <div
                class="css-1o76smw"
              >
                94.44
                <span
                  class="css-n3629o"
                >
                  %
                </span>
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Builds
              </div>
              <div
                class="css-1o76smw"
              >
                18
                <span
                  class="css-n3629o"
                >
                  / week
                </span>
              </div>
            </div>
            <div
              class="css-1gtb97y"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-sc-ea9ulj-0 jvZCGA"
                color="#05f076"
                fill="currentColor"
                focusable="false"
                height="24"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 25px; position: relative;"
      >
        <div
          class="css-tg1f9m"
        >
          <div
            class="css-hg9guf"
          >
            <img
              class="css-18ewgrp"
              src="https://img.icons8.com/color-glass/96/swamp.png"
            />
          </div>
          <div
            style="width: calc(50% - 40px);"
          >
            <a
              href="/project/q1j0ho1na7uo00u4iceqp7ot"
              style="color: rgb(255, 255, 255);"
            >
              <div
                class="css-14zx4vh"
              >
                <div>
                  flamboyant-visvesvaraya
                </div>
              </div>
              <div
                class="css-dz1vxl"
              >
                /home/tsirysndr/Documents/github/go-pipeline/example
              </div>
            </a>
            <div />
          </div>
          <div
            class="css-mxqn1d"
          >
            <div
              class="css-12j1nmi"
            >
              <div
                style="position: relative; width: 179px;"
              >
                <div
                  class="css-2cfmy6"
                  max="42.553915127154674"
                />
                <div
                  class="css-175nrtw"
                  max="71.76934374275335"
                />
                <div
                  class="css-yr5a50"
                  max="18.871453969235525"
                />
                <div
                  class="css-1tdsnrz"
                  max="22.040658576176856"
                />
                <div
                  class="css-1dsc6h1"
                  max="18.300997139986087"
                />
                <div
                  class="css-16jq5w4"
                  max="17.311586921233673"
                />
                <div
                  class="css-v1oiha"
                  max="55.69297364149339"
                />
                <div
                  class="css-it3bse"
                  max="18.71531266908866"
                />
                <div
                  class="css-1l1l3sh"
                  max="17.54966375512097"
                />
                <div
                  class="css-kpqtvv"
                  max="100"
                />
                <div
                  class="css-7z90q8"
                  max="66.09105665919456"
                />
                <div
                  class="css-x22cpr"
                  max="51.842003555692976"
                />
                <div
                  class="css-xebo26"
                  max="99.64906856303625"
                />
                <div
                  class="css-aab86p"
                  max="80.10512483574244"
                />
                <div
                  class="css-151o5yq"
                  max="43.27278348921698"
                />
                <div
                  class="css-1998adx"
                  max="62.95277112158924"
                />
                <div
                  class="css-1tampns"
                  max="36.88490376439669"
                />
                <div
                  class="css-dfhvqj"
                  max="61.68354332534591"
                />
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Speed
              </div>
              <div
                class="css-1o76smw"
              >
                31.81
                <span
                  class="css-n3629o"
                >
                  s
                </span>
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Reliability
              </div>
              <div
                class="css-1o76smw"
              >
                100
                <span
                  class="css-n3629o"
                >
                  %
                </span>
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Builds
              </div>
              <div
                class="css-1o76smw"
              >
                18
                <span
                  class="css-n3629o"
                >
                  / week
                </span>
              </div>
            </div>
            <div
              class="css-1gtb97y"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-sc-ea9ulj-0 jvZCGA"
                color="#05f076"
                fill="currentColor"
                focusable="false"
                height="24"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 25px; position: relative;"
      >
        <div
          class="css-tg1f9m"
        >
          <div
            class="css-hg9guf"
          >
            <img
              class="css-18ewgrp"
              src="https://img.icons8.com/color-glass/96/music.png"
            />
          </div>
          <div
            style="width: calc(50% - 40px);"
          >
            <a
              href="/project/ccogwb1oau3y9oav8zwbg9zk"
              style="color: rgb(255, 255, 255);"
            >
              <div
                class="css-14zx4vh"
              >
                <div>
                  admiring-meninsky
                </div>
              </div>
              <div
                class="css-dz1vxl"
              >
                /home/tsirysndr/Documents/github/gleam-pipeline/example
              </div>
            </a>
            <div />
          </div>
          <div
            class="css-mxqn1d"
          >
            <div
              class="css-12j1nmi"
            >
              <div
                style="position: relative; width: 179px;"
              >
                <div
                  class="css-11ziqnm"
                  max="99.15589967127049"
                />
                <div
                  class="css-koy087"
                  max="76.64214240129084"
                />
                <div
                  class="css-fo2lbq"
                  max="61.57477091285657"
                />
                <div
                  class="css-1bq5z1u"
                  max="53.792430792202005"
                />
                <div
                  class="css-6w4gsu"
                  max="88.2741098414151"
                />
                <div
                  class="css-7jjg6r"
                  max="100"
                />
                <div
                  class="css-1avchzx"
                  max="47.3177839054993"
                />
                <div
                  class="css-ud83k7"
                  max="51.90073091141161"
                />
                <div
                  class="css-1az3f95"
                  max="85.01691813069708"
                />
                <div
                  class="css-8qj3va"
                  max="55.746745818632824"
                />
                <div
                  class="css-ub5gh2"
                  max="41.747444218334195"
                />
                <div
                  class="css-tfwmow"
                  max="28.375498211855938"
                />
                <div
                  class="css-cqzaht"
                  max="60.84265536383012"
                />
                <div
                  class="css-httqk2"
                  max="50.961503726805304"
                />
                <div
                  class="css-1p7w21o"
                  max="66.95846930051657"
                />
                <div
                  class="css-g7wy5t"
                  max="48.03183739328332"
                />
                <div
                  class="css-1qzv6c"
                  max="82.31242549399738"
                />
                <div
                  class="css-1a3m6vp"
                  max="49.59239948462918"
                />
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Speed
              </div>
              <div
                class="css-1o76smw"
              >
                52.98
                <span
                  class="css-n3629o"
                >
                  s
                </span>
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Reliability
              </div>
              <div
                class="css-1o76smw"
              >
                100
                <span
                  class="css-n3629o"
                >
                  %
                </span>
              </div>
            </div>
            <div
              class="css-1j8ni2y"
            >
              <div>
                Builds
              </div>
              <div
                class="css-1o76smw"
              >
                20
                <span
                  class="css-n3629o"
                >
                  / week
                </span>
              </div>
            </div>
            <div
              class="css-1gtb97y"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-sc-ea9ulj-0 jvZCGA"
                color="#05f076"
                fill="currentColor"
                focusable="false"
                height="24"
                viewBox="0 0 16 16"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="height: 100px;"
    />
  </div>
</div>
`;
